src/openforms/forms/api/serializers/logic/action_serializers.py

```diff
@@ -189,6 +192,20 @@ def validate(self, data: dict) -> dict:
                 code="blank",
             )
 
+        # validate format of value for date variable
+        if action_type and action_type == LogicActionTypes.variable:
```

Comment:
```suggestion
        if action_type == LogicActionTypes.variable:
```
tautology - if action type is `None` or falsy, it will never be equal to the literal string so you can shortcut this.